test/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+function(create_test_config)
+  set(abs_top_srcdir ${PROJECT_SOURCE_DIR})
+  set(abs_top_builddir ${PROJECT_BINARY_DIR})
+  set(EXEEXT ${CMAKE_EXECUTABLE_SUFFIX})
+
+  macro(set_configure_variable var conf_var)
+    if(${var})
+      set(${conf_var}_TRUE "")
+    else()
+      set(${conf_var}_TRUE "#")
+    endif()
+  endmacro()
+
+  set_configure_variable(ENABLE_WALLET ENABLE_WALLET)
+  set_configure_variable(WITH_SQLITE USE_SQLITE)
+  set_configure_variable(WITH_BDB USE_BDB)
+  set_configure_variable(BUILD_CLI BUILD_BITCOIN_CLI)
+  set_configure_variable(BUILD_UTIL BUILD_BITCOIN_UTIL)
+  set_configure_variable(BUILD_WALLET_TOOL BUILD_BITCOIN_WALLET)
+  set_configure_variable(BUILD_DAEMON BUILD_BITCOIND_TRUE)
+  set_configure_variable(WITH_ZMQ ENABLE_ZMQ)
+  set_configure_variable(ENABLE_EXTERNAL_SIGNER ENABLE_EXTERNAL_SIGNER)
```

Comment:
Afaict these last three are still missing, right?

Reply:
The `ENABLE_EXTERNAL_SIGNER` an `USE_SYSCALL_SANDBOX` variables are not introduced in this PR. As for `ENABLE_TRACING`, see https://github.com/hebasto/bitcoin/blob/2d8930e14e2f794a03d3438e5061cfef20603284/cmake/optional.cmake#L90-L108

Anyway, the processing of the  `test/config.ini.in` file expects all three variables set:
- `ENABLE_EXTERNAL_SIGNER_TRUE`
- `ENABLE_SYSCALL_SANDBOX_TRUE`
- `ENABLE_USDT_TRACEPOINTS_TRUE`